Modify rule S7027: Update description and align with S7091 (#4319)
SonarSource/rspec: rules/S7027/java/rule.adoc

> This rule reports circular dependencies between classes that all reside in the same package.

Reviewer: This rule identifies circular dependencies between classes within the same package.
```suggestion
This rule identifies circular dependencies between classes within the same package.
```

== Why is this an issue?

> Circular dependencies occur when two classes reference each other directly or indirectly.

Reviewer: ```suggestion
Circular dependencies occur when two classes reference each other, either directly or indirectly.
```

This means that the classes' dependency structure does not represent an intuitive hierarchy, which makes it harder to understand and maintain.

=== What is the potential impact?

Circular dependencies increase the architectural complexity of the code, reducing its readability, extensibility, and maintainability.

> As a project grows, circular dependencies tend to be a seed for additional circular dependencies and further increase in architectural complexity that adds to the technical depth.

Reviewer: ```suggestion
As the project grows, circular dependencies often lead to more circular dependencies, further complicating the architecture and increasing technical debt.
```

Over time, untangling these dependencies becomes increasingly difficult.

=== Exceptions

> Circular dependencies between an inner class and its enclosing class, or between inner classes enclosed by the same class, are not reported.

Reviewer: ```suggestion
Circular dependencies between an inner class and its outer class, or between inner classes within the same enclosing class, are not flagged as issues.
```
